File: Demo/Sources/ViewControllers/Button/ButtonViewController.m
//
//  Globus
//

#import "ButtonViewController.h"

@interface ButtonViewController ()

@property(nonatomic, weak) IBOutlet GLBButton* fillButton;
@property(nonatomic, weak) IBOutlet GLBButton* strokeButton;
@property(nonatomic, weak) IBOutlet GLBButton* tintFillButton;
@property(nonatomic, weak) IBOutlet GLBButton* tintStrokeButton;

@property(nonatomic) GLBTextStyle* fillNormalStyle;
@property(nonatomic) GLBTextStyle* fillHighlightedStyle;
@property(nonatomic) GLBTextStyle* strokeNormalStyle;
@property(nonatomic) GLBTextStyle* strokeHighlightedStyle;

@end

@implementation ButtonViewController

#pragma mark - Synthesize

@synthesize fillNormalStyle = _fillNormalStyle;
@synthesize fillHighlightedStyle = _fillHighlightedStyle;
@synthesize strokeNormalStyle = _strokeNormalStyle;
@synthesize strokeHighlightedStyle = _strokeHighlightedStyle;

#pragma mark - UIViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    
    _fillButton.contentEdgeInsets = UIEdgeInsetsMake(4, 4, 4, 4);
    _fillButton.normalTitleStyle = self.fillNormalStyle;
    _fillButton.normalBackgroundColor = UIColor.lightGrayColor;
    _fillButton.normalCornerRadius = 4.0f;
    _fillButton.highlightedTitleStyle = self.fillHighlightedStyle;
    _fillButton.highlightedBackgroundColor = UIColor.darkGrayColor;
    _fillButton.highlightedCornerRadius = 8.0f;
    
    _strokeButton.contentEdgeInsets = UIEdgeInsetsMake(4, 4, 4, 4);
    _strokeButton.normalTitleStyle = self.strokeNormalStyle;
    _strokeButton.normalBorderWidth = 1.0f;
    _strokeButton.normalBorderColor = UIColor.lightGrayColor;
    _strokeButton.normalCornerRadius = 4.0f;
    _strokeButton.highlightedTitleStyle = self.strokeHighlightedStyle;
    _strokeButton.highlightedBorderColor = UIColor.darkGrayColor;
    _strokeButton.highlightedBorderWidth = 2.0f;
    _strokeButton.highlightedCornerRadius = 8.0f;
    
    _tintFillButton.contentEdgeInsets = UIEdgeInsetsMake(4, 4, 4, 4);
    _tintFillButton.glb_normalImage = [UIImage glb_imageNamed:@"ButtonIcon" renderingMode:UIImageRenderingModeAlwaysTemplate];
    _tintFillButton.normalTitleStyle = self.fillNormalStyle;
    _tintFillButton.normalBackgroundColor = UIColor.lightGrayColor;
    _tintFillButton.normalCornerRadius = 4.0f;
    _tintFillButton.normalTintColor = UIColor.darkGrayColor;
    _tintFillButton.highlightedTitleStyle = self.fillHighlightedStyle;
    _tintFillButton.highlightedBackgroundColor = UIColor.darkGrayColor;
    _tintFillButton.highlightedCornerRadius = 8.0f;
    _tintFillButton.highlightedTintColor = UIColor.blackColor;
    
    _tintStrokeButton.contentEdgeInsets = UIEdgeInsetsMake(4, 4, 4, 4);
    _tintStrokeButton.glb_normalImage = [UIImage glb_imageNamed:@"ButtonIcon" renderingMode:UIImageRenderingModeAlwaysTemplate];
    _tintStrokeButton.normalTitleStyle = self.strokeNormalStyle;
    _tintStrokeButton.normalBorderWidth = 1.0f;
    _tintStrokeButton.normalBorderColor = UIColor.lightGrayColor;
    _tintStrokeButton.normalCornerRadius = 4.0f;
    _tintStrokeButton.normalTintColor = UIColor.lightGrayColor;
    _tintStrokeButton.highlightedTitleStyle = self.strokeHighlightedStyle;
    _tintStrokeButton.highlightedBorderColor = UIColor.darkGrayColor;
    _tintStrokeButton.highlightedBorderWidth = 2.0f;
    _tintStrokeButton.highlightedCornerRadius = 8.0f;
    _tintStrokeButton.highlightedTintColor = UIColor.darkGrayColor;
}

#pragma mark - Property

- (GLBTextStyle*)fillNormalStyle {
    if(_fillNormalStyle == nil) {
        _fillNormalStyle = [GLBTextStyle new];
        _fillNormalStyle.font = [UIFont boldSystemFontOfSize:16];
        _fillNormalStyle.color = UIColor.darkGrayColor;
    }
    return _fillNormalStyle;
}

- (GLBTextStyle*)fillHighlightedStyle {
    if(_fillHighlightedStyle == nil) {
        _fillHighlightedStyle = [GLBTextStyle new];
        _fillHighlightedStyle.font = [UIFont boldSystemFontOfSize:16];
        _fillHighlightedStyle.color = UIColor.blackColor;
    }
    return _fillHighlightedStyle;
}

- (GLBTextStyle*)strokeNormalStyle {
    if(_strokeNormalStyle == nil) {
        _strokeNormalStyle = [GLBTextStyle new];
        _strokeNormalStyle.font = [UIFont italicSystemFontOfSize:16];
        _strokeNormalStyle.color = UIColor.lightGrayColor;
    }
    return _strokeNormalStyle;
}

- (GLBTextStyle*)strokeHighlightedStyle {
    if(_strokeHighlightedStyle == nil) {
        _strokeHighlightedStyle = [GLBTextStyle new];
        _strokeHighlightedStyle.font = [UIFont italicSystemFontOfSize:16];
        _strokeHighlightedStyle.color = UIColor.darkGrayColor;
    }
    return _strokeHighlightedStyle;
}

#pragma mark - Actions

- (IBAction)pressedMenu:(id)sender {
    [self.glb_slideViewController showLeftViewControllerAnimated:YES complete:nil];
}

@end

File: Demo/Sources/ViewControllers/Button/ButtonViewController.swift
//
//  Globus
//

import Globus

class ButtonViewController: GLBViewController {
    
    // MARK - Outlet property
    
    @IBOutlet fileprivate weak var fillButton: GLBButton!
    @IBOutlet fileprivate weak var strokeButton: GLBButton!
    @IBOutlet fileprivate weak var tintFillButton: GLBButton!
    @IBOutlet fileprivate weak var tintStrokeButton: GLBButton!
    
    fileprivate lazy var fillNormalStyle: GLBTextStyle = {
        let style = GLBTextStyle()
        style.font = UIFont.boldSystemFont(ofSize: 16)
        style.color = UIColor.darkGray
        return style
    }()
    fileprivate lazy var fillHighlightedStyle: GLBTextStyle = {
        let style = GLBTextStyle()
        style.font = UIFont.boldSystemFont(ofSize: 16)
        style.color = UIColor.black
        return style
    }()
    fileprivate lazy var strokeNormalStyle: GLBTextStyle = {
        let style = GLBTextStyle()
        style.font = UIFont.italicSystemFont(ofSize: 16)
        style.color = UIColor.lightGray
        return style
    }()
    fileprivate lazy var strokeHighlightedStyle: GLBTextStyle = {
        let style = GLBTextStyle()
        style.font = UIFont.italicSystemFont(ofSize: 16)
        style.color = UIColor.darkGray
        return style
    }()
    
    // MARK - UIViewController
    
    override func viewDidLoad() {
        super.viewDidLoad()
        
        self.fillButton.contentEdgeInsets = UIEdgeInsets.init(top: 4, left: 4, bottom: 4, right: 4)
        self.fillButton.normalTitleStyle = self.fillNormalStyle
        self.fillButton.normalBackgroundColor = UIColor.lightGray
        self.fillButton.normalCornerRadius = 4.0
        self.fillButton.highlightedTitleStyle = self.fillHighlightedStyle
        self.fillButton.highlightedBackgroundColor = UIColor.darkGray
        self.fillButton.highlightedCornerRadius = 8.0
        
        self.strokeButton.contentEdgeInsets = UIEdgeInsets.init(top: 4, left: 4, bottom: 4, right: 4)
        self.strokeButton.normalTitleStyle = self.strokeNormalStyle
        self.strokeButton.normalBorderWidth = 1.0
        self.strokeButton.normalBorderColor = UIColor.lightGray
        self.strokeButton.normalCornerRadius = 4.0
        self.strokeButton.highlightedTitleStyle = self.strokeHighlightedStyle
        self.strokeButton.highlightedBorderColor = UIColor.darkGray
        self.strokeButton.highlightedBorderWidth = 2.0
        self.strokeButton.highlightedCornerRadius = 8.0
        
        self.tintFillButton.contentEdgeInsets = UIEdgeInsets.init(top: 4, left: 4, bottom: 4, right: 4)
        self.tintFillButton.glb_normalImage = UIImage.glb_imageNamed("ButtonIcon", renderingMode: .alwaysTemplate)
        self.tintFillButton.normalTitleStyle = self.fillNormalStyle
        self.tintFillButton.normalBackgroundColor = UIColor.lightGray
        self.tintFillButton.normalCornerRadius = 4.0
        self.tintFillButton.normalTintColor = UIColor.darkGray
        self.tintFillButton.highlightedTitleStyle = self.fillHighlightedStyle
        self.tintFillButton.highlightedBackgroundColor = UIColor.darkGray
        self.tintFillButton.highlightedCornerRadius = 8.0
        self.tintFillButton.highlightedTintColor = UIColor.black
        
        self.tintStrokeButton.contentEdgeInsets = UIEdgeInsets.init(top: 4, left: 4, bottom: 4, right: 4)
        self.tintStrokeButton.glb_normalImage = UIImage.glb_imageNamed("ButtonIcon", renderingMode: .alwaysTemplate)
        self.tintStrokeButton.normalTitleStyle = self.strokeNormalStyle
        self.tintStrokeButton.normalBorderWidth = 1.0
        self.tintStrokeButton.normalBorderColor = UIColor.lightGray
        self.tintStrokeButton.normalCornerRadius = 4.0
        self.tintStrokeButton.normalTintColor = UIColor.lightGray
        self.tintStrokeButton.highlightedTitleStyle = self.strokeHighlightedStyle;
        self.tintStrokeButton.highlightedBorderColor = UIColor.darkGray
        self.tintStrokeButton.highlightedBorderWidth = 2.0
        self.tintStrokeButton.highlightedCornerRadius = 8.0
        self.tintStrokeButton.highlightedTintColor = UIColor.darkGray
    }
    
    // MARK - Action
    
    @IBAction internal func pressedMenu(_ sender: Any) {
        self.glb_slideViewController?.showLeftViewController(animated: true, complete: nil)
    }
    
    // MARK: - GLBNibExtension
    
    public override static func nibName() -> String {
        return "ButtonViewController-Swift"
    }
    
}
